vscode/src/serialView.ts

```diff
@@ -35,9 +37,17 @@ export class SerialProvider implements TreeDataProvider<TreeItem> {
     if (element) {
       if (element instanceof SerialPort) {
         const info = await getSerialInfo(this.context, element);
-        if (info) return [info];
+        if (info === SerialStatus.Busy) {
+          const cachedValue = this.infoCache.get(element.name);
+          return cachedValue ? [cachedValue] : [];
+        }
+
+        if (info instanceof SerialInfo) {
+          this.infoCache.set(element.name, info);
+          return [info];
+        }
+        this.infoCache.delete(element.name);
```

Comment:
Is this, because you assume that it's `Disconnected` ?
If yes, add a comment, or even an assert (although I'm not sure asserts exist in JS).